File: app/src/main/java/com/imangazalievm/bubbble/data/repository/CommentsRepository.java
package com.imangazalievm.bubbble.data.repository;

import com.imangazalievm.bubbble.data.network.DribbbleApiService;
import com.imangazalievm.bubbble.domain.models.Comment;
import com.imangazalievm.bubbble.domain.models.ShotCommentsRequestParams;
import com.imangazalievm.bubbble.domain.repository.ICommentsRepository;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.Single;

public class CommentsRepository implements ICommentsRepository {

    private DribbbleApiService dribbbleApiService;

    @Inject
    public CommentsRepository(DribbbleApiService dribbbleApiService) {
        this.dribbbleApiService = dribbbleApiService;
    }

    public Single<List<Comment>> getComments(ShotCommentsRequestParams requestParams) {
        return dribbbleApiService.getShotComments(requestParams.getShotId(), requestParams.getPage(), requestParams.getPageSize());
    }

}

File: app/src/main/java/com/bubbble/data/shots/CommentsRepository.kt
package com.bubbble.data.shots

import com.bubbble.core.models.Comment
import com.bubbble.core.models.shot.ShotCommentsParams
import com.bubbble.core.network.DribbbleApi
import javax.inject.Inject

class CommentsRepository @Inject constructor(private val dribbbleApi: DribbbleApi) {

    suspend fun getComments(
        requestParams: ShotCommentsParams
    ): List<Comment> {
        return dribbbleApi.getShotComments(
            requestParams.shotId,
            requestParams.page,
            requestParams.pageSize
        )
    }

}